src/app/icd/ICDMonitoringTable.cpp

```diff
@@ -72,6 +72,10 @@ CHIP_ERROR ICDMonitoringEntry::Deserialize(TLV::TLVReader & reader)
             case to_underlying(Fields::kKey): {
                 ByteSpan buf(key.AsMutable<Crypto::Aes128KeyByteArray>());
                 ReturnErrorOnFailure(reader.Get(buf));
+                // Since we are storing either the raw key or a key ID, we must
+                // simply copy the data as is in the keyHandle.
+                // Calling SetKey here would create another key in storage and will cause
+                // key leakage in some implementation.
```

Comment:
mentioning or at least giving an example of which implementations (PSA or not PSA for example) would be beneficial here.

Reply:
That's what I did in an earlier PR but it was pointed out by Boris that it won't be future proof if we are anchoring ourselves to these two option only. Keeping it vague allow us to be future proof should future Crypto implementation be added to Matter.

It also draw a line between the classes. The ICDMonitoringTable shouldn't be aware of what implementation is actually used.